routing/base/astar_algorithm.hpp

```diff
@@ -349,6 +369,14 @@ typename AStarAlgorithm<TGraph>::Result AStarAlgorithm<TGraph>::FindPathBidirect
         if (!foundAnyPath || bestPathReducedLength > curPathReducedLength)
         {
           bestPathReducedLength = curPathReducedLength;
+          bestPathRealLength = stateV.distance + len - pV + distW;
+          bestPathRealLength += cur->pS;
+
+          if (distW > 0)
```

Comment:
Why? It's guaranteed that stateW is from the second wave and it's not a problem if it will coincide with the starting vertex of the second wave. You'll just extract pi(s) from pi(s) and ll get zero.